src/main/java/aspects/LoggingAspect.java

```diff
@@ -29,33 +34,80 @@
             "&& !@annotation(annotations.NoLogging)")
     public Object logRequest(ProceedingJoinPoint joinPoint) throws Throwable {
-        MethodSignature ms = (MethodSignature) joinPoint.getSignature();
-        Method m = ms.getMethod();
-        Object requestBean = null;
-        final String requestPath = m.getAnnotation(RequestMapping.class).value()[0]; //Should be only one
-        try {
-            requestBean = joinPoint.getArgs()[0];
-            log.info("Request to " + requestPath + " with bean " + requestBean);
-        } catch(ArrayIndexOutOfBoundsException e) {
-            // no request body
-            log.info("Request to " + requestPath + " with no request body.");
-        }
+        final String requestPath = this.getRequestPath(joinPoint);
+        final Object requestBean = this.getRequestBean(joinPoint);
+        final String requestBody = this.getRequestBody(joinPoint);
+        final AuthenticatedRequestBean authenticatedRequestBean =
+                (requestBean != null && requestBean instanceof AuthenticatedRequestBean) ?
+                        (AuthenticatedRequestBean) requestBean : null;
+
+        String domain = null;
+        String username = null;
+        String restoreAs = null;
+
+        if (authenticatedRequestBean != null) {
+            domain = authenticatedRequestBean.getDomain();
+            username = authenticatedRequestBean.getUsername();
+            restoreAs = authenticatedRequestBean.getRestoreAs();
+            String bean = authenticatedRequestBean.toString();
 
-        if (requestBean != null && requestBean instanceof AuthenticatedRequestBean) {
-            final AuthenticatedRequestBean authenticatedRequestBean = (AuthenticatedRequestBean) requestBean;
             raven.newBreadcrumb()
                     .setData(
                             "path", requestPath,
-                            "domain", authenticatedRequestBean.getDomain(),
-                            "username", authenticatedRequestBean.getUsername(),
-                            "restoreAs", authenticatedRequestBean.getRestoreAs(),
-                            "bean", authenticatedRequestBean.toString()
+                            "domain", domain,
+                            "username", username,
+                            "restoreAs", restoreAs,
+                            "bean", bean
                     )
                     .record();
         }
 
         Object result = joinPoint.proceed();
-        log.info("Request to " + requestPath + " returned result " + result);
+
+        JSONObject logLine = (new JSONObject())
+                .put("request_path", requestPath)
+                .put("request_body", requestBody)
+                .put("project_space", domain)
+                .put("username", username)
+                .put("restore_as", restoreAs)
+                // TODO: Figure out a way to get response JSON.
+                .put("responseBean", result);
```

Comment:
According to https://stackoverflow.com/questions/11692303/turning-a-bean-to-json-in-java you can use

```
new JSONObject(result)
```

Not sure how much I trust that, but it's worth a shot

Reply:
This is pretty fraught, because Spring uses the incoming requested content type to determine the serialization of downstream body. Provided an alternative above.